Test failures on latest devel

With latest devel branches of libyang and sysrepo (CESNET/libyang@073db02ea0ef2990982fd379144cfd02c054bc0b and sysrepo/sysrepo@e23f417037fb55a047633e02ebc7f9a945fc8c92), the test suite fails:

```
+ ctest3 --output-on-failure
Test project /home/turbo-hipster/jobs/cf564b27103c/22/922/1/check/check-czechlight-devtoolset7-el7/1fa3006/build-sysrepo
      Start  1: common_test
 1/15 Test  #1: common_test ......................***Failed    0.09 sec
[==========] Running 19 test(s).
[ RUN      ] sr_llist_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_llist_test
[ RUN      ] sr_list_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[ERR] (sr_list_rm_at:186) Index 94 out of bounds of the list (0 - 93)
[ERR] (sr_list_rm_at:186) Index 100 out of bounds of the list (0 - 93)
[       OK ] sr_list_test
[ RUN      ] sr_ordered_list_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_ordered_list_test
[ RUN      ] circular_buffer_test1
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[DBG] (sr_cbuff_init:457) Initiating circular buffer for 2 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=1.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=2.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 2 to 4 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=3.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=3, current count=4.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=1, count=3.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=2, count=2.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=3.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=4.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 4 to 8 elements.
[DBG] (sr_cbuff_enqueue:505) Moving 2 circular buffer elements from pos 0 to pos 4.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=6, current count=5.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=7, current count=6.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=7.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=8.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=3, count=7.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=4, count=6.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=5, count=5.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=6, count=4.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=5.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=3, current count=6.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=4, current count=7.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=5, current count=8.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 8 to 16 elements.
[DBG] (sr_cbuff_enqueue:505) Moving 6 circular buffer elements from pos 0 to pos 8.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=14, current count=9.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=15, current count=10.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=11.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=12.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=13.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=3, current count=14.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=4, current count=15.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=5, current count=16.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 16 to 32 elements.
[DBG] (sr_cbuff_enqueue:505) Moving 6 circular buffer elements from pos 0 to pos 16.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=22, current count=17.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=23, current count=18.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=24, current count=19.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=25, current count=20.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=26, current count=21.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=27, current count=22.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=28, current count=23.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=29, current count=24.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=30, current count=25.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=31, current count=26.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=27.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=28.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=29.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=3, current count=30.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=4, current count=31.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=5, current count=32.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 32 to 64 elements.
[DBG] (sr_cbuff_enqueue:505) Moving 6 circular buffer elements from pos 0 to pos 32.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=38, current count=33.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=39, current count=34.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=40, current count=35.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=41, current count=36.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=42, current count=37.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=43, current count=38.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=44, current count=39.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=45, current count=40.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=46, current count=41.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=47, current count=42.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=48, current count=43.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=49, current count=44.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=7, count=43.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=8, count=42.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=9, count=41.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=10, count=40.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=11, count=39.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=12, count=38.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=13, count=37.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=14, count=36.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=15, count=35.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=16, count=34.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=17, count=33.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=18, count=32.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=19, count=31.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=20, count=30.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=21, count=29.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=22, count=28.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=23, count=27.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=24, count=26.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=25, count=25.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=26, count=24.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=27, count=23.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=28, count=22.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=29, count=21.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=30, count=20.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=31, count=19.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=32, count=18.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=33, count=17.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=34, count=16.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=35, count=15.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=36, count=14.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=37, count=13.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=38, count=12.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=39, count=11.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=40, count=10.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=41, count=9.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=42, count=8.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=43, count=7.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=44, count=6.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=45, count=5.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=46, count=4.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=47, count=3.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=48, count=2.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=49, count=1.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=50, count=0.
[       OK ] circular_buffer_test1
[ RUN      ] circular_buffer_test2
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[DBG] (sr_cbuff_init:457) Initiating circular buffer for 2 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=1.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=2.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 2 to 4 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=3.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=3, current count=4.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 4 to 8 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=4, current count=5.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=5, current count=6.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=6, current count=7.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=1, count=6.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=2, count=5.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=3, count=4.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=7, current count=5.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=6.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=7.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=8.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 8 to 16 elements.
[DBG] (sr_cbuff_enqueue:505) Moving 3 circular buffer elements from pos 0 to pos 8.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=11, current count=9.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=12, current count=10.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=13, current count=11.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=14, current count=12.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=15, current count=13.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=14.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=15.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=16.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 16 to 32 elements.
[DBG] (sr_cbuff_enqueue:505) Moving 3 circular buffer elements from pos 0 to pos 16.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=19, current count=17.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=4, count=16.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=5, count=15.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=6, count=14.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=7, count=13.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=8, count=12.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=9, count=11.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=10, count=10.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=11, count=9.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=12, count=8.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=13, count=7.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=14, count=6.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=15, count=5.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=16, count=4.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=17, count=3.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=18, count=2.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=19, count=1.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=20, count=0.
[       OK ] circular_buffer_test2
[ RUN      ] circular_buffer_test3
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[DBG] (sr_cbuff_init:457) Initiating circular buffer for 2 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=1.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=2.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 2 to 4 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=3.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=3, current count=4.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=1, count=3.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=2, count=2.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=3, count=1.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=0, count=0.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=0, current count=1.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=1, current count=2.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=3.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=3, current count=4.
[DBG] (sr_cbuff_enqueue:497) Enlarging circular buffer from 4 to 8 elements.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=4, current count=5.
[DBG] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=5, current count=6.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=1, count=5.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=2, count=4.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=3, count=3.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=4, count=2.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=5, count=1.
[DBG] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=6, count=0.
[       OK ] circular_buffer_test3
[ RUN      ] sr_bitset_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[ERR] (sr_bitset_init:883) NULL value detected for bitset_p argument of sr_bitset_init
[       OK ] sr_bitset_test
[ RUN      ] logger_callback_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[DBG] (logger_callback_test:622) Testing logging callback 5, 4, 3, ...
[INF] (logger_callback_test:623) Testing logging callback 2, 1, 0, GO!
LOG level=4: (logger_callback_test:622) Testing logging callback 5, 4, 3, ...
LOG level=3: (logger_callback_test:623) Testing logging callback 2, 1, 0, GO!
[       OK ] logger_callback_test
[ RUN      ] sr_locking_set_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[DBG] (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
[INF] (sr_locking_set_lock_file_open:701) File /tmp/testing_file locked by other process
[DBG] (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
[ERR] (sr_locking_set_unlock_close_file:819) File /tmp/testing_file has not been locked in this context
[DBG] (sr_locking_set_lock_fd:789) File /tmp/testing_file has been locked
[INF] (sr_locking_set_lock_fd:779) File /tmp/testing_file can not be locked
[DBG] (sr_locking_set_unlock_close_fd:858) File /tmp/testing_file (fd = 14) has been unlocked
[DBG] (sr_locking_set_lock_fd:789) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_fd:858) File /tmp/testing_file (fd = 15) has been unlocked
[DBG] (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_fd:858) File /tmp/testing_file (fd = 14) has been unlocked
[DBG] (sr_locking_set_lock_fd:789) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
[DBG] (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
[DBG] (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
[DBG] (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
[DBG] (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
[DBG] (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
[DBG] (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
LOG level=4: (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
LOG level=3: (sr_locking_set_lock_file_open:701) File /tmp/testing_file locked by other process
LOG level=4: (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
LOG level=1: (sr_locking_set_unlock_close_file:819) File /tmp/testing_file has not been locked in this context
LOG level=4: (sr_locking_set_lock_fd:789) File /tmp/testing_file has been locked
LOG level=3: (sr_locking_set_lock_fd:779) File /tmp/testing_file can not be locked
LOG level=4: (sr_locking_set_unlock_close_fd:858) File /tmp/testing_file (fd = 14) has been unlocked
LOG level=4: (sr_locking_set_lock_fd:789) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_fd:858) File /tmp/testing_file (fd = 15) has been unlocked
LOG level=4: (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_fd:858) File /tmp/testing_file (fd = 14) has been unlocked
LOG level=4: (sr_locking_set_lock_fd:789) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
LOG level=4: (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
LOG level=4: (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
LOG level=4: (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
LOG level=4: (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
LOG level=4: (sr_locking_set_lock_file_open:725) File /tmp/testing_file has been locked
LOG level=4: (sr_locking_set_unlock_close_file:832) File /tmp/testing_file has been unlocked
[       OK ] sr_locking_set_test
[ RUN      ] sr_node_t_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
<container xmlns="urn:ietf:params:xml:ns:yang:example">
  <list>
    <key1>key1</key1>
    <key2>key2</key2>
    <leaf>leaf12</leaf>
  </list>
  <list>
    <key1>keyA</key1>
    <key2>keyB</key2>
    <leaf>leafAB</leaf>
  </list>
</container>
<number xmlns="urn:ietf:params:xml:ns:yang:example">2</number>
<number xmlns="urn:ietf:params:xml:ns:yang:example">1</number>
<number xmlns="urn:ietf:params:xml:ns:yang:example">42</number>
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_node_t_test
[ RUN      ] sr_node_t_with_augments_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
<item xmlns="urn:ietf:params:xml:ns:yang:small-module">
  <name>hey hou</name>
  <info xmlns="urn:ietf:params:xml:ns:yang:info">info 123</info>
</item>
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_node_t_with_augments_test
[ RUN      ] sr_node_t_rpc_input_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
libyang[0]: Must condition ". != /top-level-default" not satisfied. (path: /test-module:activate-software-image/image-name)
[  ERROR   ] --- 0 != 0x1
[   LINE   ] --- /home/turbo-hipster/git/cf564b27103c/CzechLight/cla-sysrepo/submodules/sysrepo/tests/common_test.c:1056: error: Failure!
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[  FAILED  ] sr_node_t_rpc_input_test
[ RUN      ] sr_node_t_rpc_output_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
<activate-software-image xmlns="urn:ietf:params:xml:ns:yang:test-module">
  <status>Installed</status>
  <init-log>
    <log-msg>
      <msg>Successfully loaded software image.</msg>
      <time>1469625110</time>
      <msg-type>debug</msg-type>
    </log-msg>
    <log-msg>
      <msg>Some soft limit exceeded...</msg>
      <time>1469625150</time>
      <msg-type>warning</msg-type>
    </log-msg>
  </init-log>
</activate-software-image>
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_node_t_rpc_output_test
[ RUN      ] sr_free_schema_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_free_schema_test
[ RUN      ] sr_error_info_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_error_info_test
[ RUN      ] sr_create_uri_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_create_uri_test
[ RUN      ] sr_get_system_groups_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
[DBG] (sr_get_system_groups_test:1368) User 'turbo-hipster' is member of the group 'turbo-hipster'.
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
LOG level=4: (sr_get_system_groups_test:1368) User 'turbo-hipster' is member of the group 'turbo-hipster'.
[       OK ] sr_get_system_groups_test
[ RUN      ] sr_free_list_of_strings_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_free_list_of_strings_test
[ RUN      ] sr_dup_data_tree_to_ctx_test
[DBG] (sr_log_stderr:141) Setting log level for stderr logs to 4.
LOG level=4: (sr_log_stderr:141) Setting log level for stderr logs to 4.
[       OK ] sr_dup_data_tree_to_ctx_test
[==========] 19 test(s) run.
[  PASSED  ] 18 test(s).
[  FAILED  ] 1 test(s), listed below:
[  FAILED  ] sr_node_t_rpc_input_test
 1 FAILED TEST(S)

```
There are other failures as well, this one is the first.

--- a/src/data_manager.c
+++ b/src/data_manager.c
@@ -1416,7 +1416,7 @@ dm_load_data_tree_file(dm_ctx_t *dm_ctx, int fd, const char *data_filename, dm_s
             ly_ctx_set_module_data_clb(tmp_ctx->ctx, dm_module_clb, dm_ctx);
 
             ly_errno = LY_SUCCESS;
-            tmp_node = lyd_parse_fd(tmp_ctx->ctx, fd, SR_FILE_FORMAT_LY, LYD_OPT_TRUSTED | LYD_OPT_CONFIG);
+            tmp_node = lyd_parse_fd(tmp_ctx->ctx, fd, SR_FILE_FORMAT_LY, LYD_OPT_TRUSTED | LYD_OPT_STRICT | LYD_OPT_CONFIG);
             md_ctx_unlock(dm_ctx->md_ctx);
 
             if (NULL == tmp_node && LY_SUCCESS != ly_errno) {
@@ -1435,7 +1435,7 @@ dm_load_data_tree_file(dm_ctx_t *dm_ctx, int fd, const char *data_filename, dm_s
         } else {
             ly_errno = LY_SUCCESS;
             /* use LYD_OPT_TRUSTED, validation will be done later */
-            data_tree = lyd_parse_fd(schema_info->ly_ctx, fd, SR_FILE_FORMAT_LY, LYD_OPT_TRUSTED | LYD_OPT_CONFIG);
+            data_tree = lyd_parse_fd(schema_info->ly_ctx, fd, SR_FILE_FORMAT_LY, LYD_OPT_TRUSTED | LYD_OPT_STRICT | LYD_OPT_CONFIG);
             if (NULL == data_tree && LY_SUCCESS != ly_errno) {
                 SR_LOG_ERR("Parsing data tree from file %s failed: %s", data_filename, ly_errmsg(schema_info->ly_ctx));
                 free(data);
@@ -6822,13 +6822,13 @@ dm_parse_event_notif(rp_ctx_t *rp_ctx, rp_session_t *session, sr_mem_ctx_t *sr_m
             rc = SR_ERR_INTERNAL;
             goto cleanup;
         }
-        data_tree = lyd_parse_xml(ly_ctx, &xml, LYD_OPT_NOTIF | LYD_OPT_TRUSTED, NULL);
+        data_tree = lyd_parse_xml(ly_ctx, &xml, LYD_OPT_NOTIF | LYD_OPT_TRUSTED | LYD_OPT_STRICT, NULL);
     } else if (notification->data_type == NP_EV_NOTIF_DATA_STRING) {
         CHECK_NULL_ARG(notification->data.string);
-        data_tree = lyd_parse_mem(ly_ctx, notification->data.string, LYD_XML, LYD_OPT_NOTIF | LYD_OPT_TRUSTED, NULL);
+        data_tree = lyd_parse_mem(ly_ctx, notification->data.string, LYD_XML, LYD_OPT_NOTIF | LYD_OPT_TRUSTED | LYD_OPT_STRICT, NULL);
     } else {
         CHECK_NULL_ARG(notification->data.string);
-        data_tree = lyd_parse_mem(ly_ctx, notification->data.string, LYD_JSON, LYD_OPT_NOTIF | LYD_OPT_TRUSTED, NULL);
+        data_tree = lyd_parse_mem(ly_ctx, notification->data.string, LYD_JSON, LYD_OPT_NOTIF | LYD_OPT_TRUSTED | LYD_OPT_STRICT, NULL);
     }
     if (NULL == data_tree) {
         SR_LOG_ERR("Error by parsing notification data: %s", ly_errmsg(ly_ctx));

--- a/src/nacm.c
+++ b/src/nacm.c
@@ -493,7 +493,7 @@ nacm_load_config(nacm_ctx_t *nacm_ctx, const sr_datastore_t ds)
     ly_ctx_set_module_data_clb(nacm_ctx->schema_info->ly_ctx, dm_module_clb, nacm_ctx->dm_ctx);
 
     ly_errno = 0;
-    data_tree = lyd_parse_fd(nacm_ctx->schema_info->ly_ctx, fd, SR_FILE_FORMAT_LY, LYD_OPT_TRUSTED | LYD_OPT_CONFIG);
+    data_tree = lyd_parse_fd(nacm_ctx->schema_info->ly_ctx, fd, SR_FILE_FORMAT_LY, LYD_OPT_TRUSTED | LYD_OPT_STRICT | LYD_OPT_CONFIG);
     if (NULL == data_tree && LY_SUCCESS != ly_errno) {
         SR_LOG_ERR("Parsing of data tree from file %s failed: %s", ds_filepath, ly_errmsg(nacm_ctx->schema_info->ly_ctx));
         goto cleanup;

--- a/tests/common_test.c
+++ b/tests/common_test.c
@@ -1053,7 +1053,7 @@ sr_node_t_rpc_input_test(void **state)
     sr_free_trees(trees, tree_cnt);
 
     /* add default nodes */
-    assert_int_equal(0, lyd_validate(&data_tree, LYD_OPT_STRICT | LYD_OPT_RPC, NULL));
+    assert_int_equal(0, lyd_validate(&data_tree, LYD_OPT_RPC | LYD_OPT_NOEXTDEPS, NULL));
 
     /* convert RPC input back to sysrepo trees */
     nodeset = lyd_find_path(data_tree, "/test-module:activate-software-image/./*");

--- a/tests/yang/cross-module.yang
+++ b/tests/yang/cross-module.yang
@@ -24,7 +24,7 @@ module cross-module {
          }
       }
 
-      must 'number = /rd:magic_number' {
+      must 'count(number) = 0 or number = /rd:magic_number' {
          error-message "Number must be equal to magic_number";
       }
    }

--- a/tests/yang/test-module.yang
+++ b/tests/yang/test-module.yang
@@ -209,11 +209,11 @@ module test-module {
     leaf ifMTU {
         type uint32;
     }
-    must "ifType != 'ethernet' or " +
+    must "count(ifType) = 0 or ifType != 'ethernet' or " +
          "(ifType = 'ethernet' and ifMTU = 1500)" {
          error-message "An ethernet MTU must be 1500";
     }
-    must "ifType != 'atm' or " +
+    must "count(ifType) = 0 or ifType != 'atm' or " +
          "(ifType = 'atm' and ifMTU <= 17966 and ifMTU >= 64)" {
          error-message "An atm MTU must be  64 .. 17966";
     }
@@ -477,7 +477,7 @@ module test-module {
         }
         leaf time-of-change {
           type uint32;
-          must '. > /rd:magic_number' {
+          must '. > /rd:magic_number or count(/rd:magic_number) = 0' {
              error-message "time-of-change must be greater than magic_number";
           }
         }

